src/sync/peer-sync-controller.js

```diff
@@ -184,10 +247,7 @@ export class PeerSyncController {
     }
     this.#log('capability %o', this.#syncCapability)
 
-    // If any namespace has new data, update what is enabled
-    if (Object.values(didUpdate).indexOf(true) > -1) {
```

Comment:
I think removing this might have significant performance implications.

Reply:
I'm not too worried about this.

`#updateEnabledNamespaces` eventually calls `#enableNamespace` and `#disableNamespace`. `#enableNamespace` won't double-enable namespaces:

https://github.com/digidem/mapeo-core-next/blob/2cb7a0b0eba0fb14e24edefc2a9a6032ef70494a/src/sync/peer-sync-controller.js#L331-L332

Same thing is true for `#disableNamespace`.

It's possible that there's something slow along the way, but I see no (1) async (2) variable-sized loops[^0] (3) external system calls.

Might be wrong, though!

[^0]: We iterate over constant-sized arrays, like `NAMESPACES`.

Reply:
oh yes sorry my mistake, I was muddling this up with the reading sync state (bitfields) which we try to minimize because that _is_ expensive. I don't see anything expensive here, and I think we throttle sync state updates, so it won't be called too often either.